### ISSUE TITLE
get_readable Association Chain Causing Silent Error

### ISSUE THREAD

**Issue by ethagnawl** [piccolo-orm/piccolo_admin#170]
This may be an issue originating with the ORM but I'll start here and will be happy to open a replacement or separate issue in the ORM repository, if the problem is real and multi-faceted.

Given the following Piccolo tables and relationships, something is going wrong on the server-side when viewing the and I see a generic "Problem fetching thing_one rows." shown in UI (500 HTTP response with "Internal Server Error" in body) and no errors shown in the server log. I'm not certain if associations are supposed to be able to be walked in this way, so I may be starting from a false premise. The interesting thing is that this _does_ work in some views (add record) but not others (list records).

```
# NOTE: This is a sanitized version of the app where I'm seeing this behavior and I haven't tried executing this as-is. I'll try to put together a functioning example.

class ThingOne(Table):
    id = UUID(primary_key=True)
    name = Text(unique=True)

class ThingTwo(Table):
    active = Boolean(default=True)
    id = UUID(primary_key=True)
    thing_one = ForeignKey(references=ThingOne, required=True)

class ThingThree(Table):
    thing_two = ForeignKey(references=ThingTwo)
    id = UUID(primary_key=True)

    @classmethod
    def get_readable(cls):
        return Readable(
            template="%s",
            columns=[
                cls.thing_two.thing_one # works
                cls.thing_two.thing_one.name # seems to break list view; works in add record view
            ],
        )

class ThingFour(Table):
    job_metadata = JSON()
    thing_three = ForeignKey(references=ThingThree, required=True)
```

I think at a minimum, the error should be shown in the logs and, if possible, it'd be wonderful if the multi-layered readable could work in all situations. It might also be good to handle the underlying error cleanly and have a more detailed error message bubble up to the client UI.

**Comment by dantownsend** [comment]
There might be an edge case which breaks readable. It does support joins - I've used it before in some of my apps.

These are the tests we have at the moment.

https://github.com/piccolo-orm/piccolo/blob/master/tests/columns/test_readable.py
https://github.com/piccolo-orm/piccolo/blob/master/tests/table/instance/test_get_related_readable.py

They don't cover every scenario, so there could be a bug.

If you try this query do you get any errors?

```
piccolo shell run
>>> await ThingThree.select(ThingThree.get_readable())
```
That way we will know if it's a problem with Piccolo, or something with Piccolo Admin / Piccolo API and how they interact with readable.

**Comment by ethagnawl** [comment]
I'll try that query and also try to write a failing test tomorrow morning and follow up.

**Comment by dantownsend** [comment]
Cool, thanks

**Comment by ethagnawl** [comment]
The queries seems to work as expected:

```
thing_ones: [{'readable': '1'}]
thing_twos:  [{'readable': '1'}]
thing_threes: [{'readable': 'three name: thing_three - two name: thing_two - one name: thing_one'}]
thing_fours: [{'readable': '1'}]
```

However, when running a modified version of the example.py script from the repository using my contrived tables, I see the following exception occur in the scenario I'd outlined (viewing the index page for "Thing Four"):

```
[2022-06-29 10:50:38 -0400] [2441278] [ERROR] Error in ASGI Framework
Traceback (most recent call last):
  File "/home/peter/.local/lib/python3.9/site-packages/hypercorn/asyncio/task_group.py", line 21, in _handle
    await invoke_asgi(app, scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/hypercorn/utils.py", line 247, in invoke_asgi
    await app(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/base.py", line 57, in __call__
    task_group.cancel_scope.cancel()
  File "/home/peter/.local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 574, in __aexit__
    raise exceptions[0]
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/base.py", line 30, in coro
    await self.app(scope, request.receive, send_stream.send)
  File "/home/peter/.local/lib/python3.9/site-packages/fastapi/applications.py", line 208, in __call__
    await super().__call__(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/routing.py", line 656, in __call__
    await route.handle(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/routing.py", line 408, in handle
    await self.app(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/authentication.py", line 48, in __call__
    await self.app(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/fastapi/applications.py", line 208, in __call__
    await super().__call__(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 181, in __call__
    raise exc
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/middleware/errors.py", line 159, in __call__
    await self.app(scope, receive, _send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/exceptions.py", line 82, in __call__
    raise exc
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/exceptions.py", line 71, in __call__
    await self.app(scope, receive, sender)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/routing.py", line 656, in __call__
    await route.handle(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/routing.py", line 259, in handle
    await self.app(scope, receive, send)
  File "/home/peter/.local/lib/python3.9/site-packages/starlette/routing.py", line 61, in app
    response = await func(request)
  File "/home/peter/.local/lib/python3.9/site-packages/fastapi/routing.py", line 226, in app
    raw_response = await run_endpoint_function(
  File "/home/peter/.local/lib/python3.9/site-packages/fastapi/routing.py", line 159, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo_api/fastapi/endpoints.py", line 123, in get
    return await piccolo_crud.root(request=request)
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo_api/crud/endpoints.py", line 517, in root
    return await self.get_all(request, params=params)
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo_api/crud/validators.py", line 76, in inner_coroutine_function
    return await function(*args, **kwargs)
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo_api/crud/endpoints.py", line 701, in get_all
    [
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo_api/crud/endpoints.py", line 702, in <listcomp>
    self.table._get_related_readable(i)
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo/table.py", line 606, in _get_related_readable
    output_column = getattr(column, fk._meta.name)
  File "/home/peter/.local/lib/python3.9/site-packages/piccolo/columns/column_types.py", line 2080, in __getattribute__
    raise AttributeError
AttributeError
```

So, I think if there is an issue here, it's with the ORM. I'll have to pick this up later but at least I've found a thread. For reference, the issue can be reproduced using [this](https://github.com/ethagnawl/piccolo_admin/blob/nested-readable-bug-170/piccolo_admin/example.py) example.py.

**Comment by dantownsend** [comment]
@ethagnawl Thanks for investigating this - you're right, it does look like an ORM problem.

**Comment by dantownsend** [comment]
@ethagnawl This should be fixed now in Piccolo 0.80.1.

**Comment by ethagnawl** [comment]
Thanks, @dantownsend! :grin:
